pkgs/applications/window-managers/icewm/default.nix

```
@@ -1,36 +1,40 @@
-{ stdenv, fetchurl, cmake, gettext
+{ stdenv, fetchFromGitHub, cmake, gettext, perl, asciidoc
 , libjpeg, libtiff, libungif, libpng, imlib, expat
 , freetype, fontconfig, pkgconfig, gdk-pixbuf
 , mkfontdir, libX11, libXft, libXext, libXinerama
 , libXrandr, libICE, libSM, libXpm, libXdmcp, libxcb
-, libpthreadstubs, pcre }:
+, libpthreadstubs, pcre, libXdamage, libXcomposite, libXfixes
+, libsndfile, fribidi }:
 
-with stdenv.lib;
 stdenv.mkDerivation rec {
-  name = "icewm-${version}";
-  version = "1.4.2";
+  pname = "icewm";
+  version = "1.6.0";
 
-  buildInputs =
-  [ cmake gettext libjpeg libtiff libungif libpng imlib expat
+  src = fetchFromGitHub {
+    owner  = "bbidulock";
+    repo   = "icewm";
+    rev    = version;
+    sha256 = "1l8hjmb19d7ds7z21cx207h86wkjcmmmamcnalgkwh4alvbawc2p";
+  };
+
+  nativeBuildInputs = [ cmake perl asciidoc ];
+
+  buildInputs = [
+    gettext libjpeg libtiff libungif libpng imlib expat
     freetype fontconfig pkgconfig gdk-pixbuf mkfontdir libX11
```

i think pkgconfig can be moved to native inputs as well.
```suggestion
    freetype fontconfig gdk-pixbuf mkfontdir libX11
```